docs: Page Content Layout guidelines (#3057)
commercetools/merchant-center-application-kit: website/src/content/development/page-content-layouts.mdx

---
title: Page Content Layouts
navLevels: 2
---

> At commercetools, one core design principle is a visual hierarchy, having a clean and usable user interface using spacings. As part of these guidelines, it's essential to properly structure a page's overall content.

Reviewer: The original sentence was more correct. For instance, visual hierarchy is not a design principle but a heuristic. I would try to keep the terminology and meaning close to the one before.

How about this? ⬇️

```suggestion
At commercetools, one of our core design heuristics is about visual hierarchy, to have a clean and usable user interface by making use of spacings. As part of these guidelines, it's essential to properly structure the page's overall content.
```

Reviewer 2: Hi @emmenko, thanks for pointing this out. I think the grammar and readability of the sentence are a bit off. How about
> At commercetools, visual hierarchy is one of our core design heuristics to create a clean and user-friendly interface by utilizing spacing. For visual hierarchy, it's crucial to structure the page content properly.

Reviewer: Yes sounds good.

Reviewer: @Anshuman71 Do you push the sentence update here?

Reviewer: ℹ️ I pushed the suggestion now. https://github.com/commercetools/merchant-center-application-kit/pull/3057/commits/097fd23945f66b0b669ce754f1808ccaff5f187f

To support various page content requirements, we provide the following layout types:

| Layout type | Use case |
|-------------|----------|
| [Narrow](#narrow) | Simple forms, collapsible panels, and cards. |
| [Wide (single column)](#narrow) | Large forms or tables with few columns. |
| [Wide (Two columns 1:1)](#narrow) | Content grouped together to save vertical space. |
| [Wide (Two columns 2:1)](#narrow) | Wider tables and some meta information. |
| [Full](#full) | Horizontal reading flow such as large tables. |

## Narrow

A narrow layout consists of a single-column with a constrained width size.

![Layout narrow](/images/page-content-layout-narrow.png)

To implement a narrow layout, use the [`<PageContentNarrow>`](/api-reference/commercetools-frontend-application-components#pagecontentnarrow) component.

```jsx
import { PageContentNarrow } from '@commercetools-frontend/application-components';

const ChannelsDetails = () => {
  // ...
  return (
    <PageContentNarrow>
      <Spacings.Stack>
        <div>Title</div>
        <ChannelsForm />
      </Spacings.Stack>
    </PageContentNarrow>
  )
};
```

> ### Best practices
> 
> * Use a narrow layout for simple forms where form fields must span across the available width (`horizontalConstraint: scale`).

Reviewer: I remember someone from the writers once told me to avoid starting a section with a bullet list.
That's why I added the short sentence `This layout should:`.

To give an example, see Polaris docs

<img width="244" alt="image" src="https://user-images.githubusercontent.com/1110551/236758644-eee92041-3afa-4a58-a6ed-9faf940ca1f0.png">

```suggestion
### Best practices

This layout should:
* Use a narrow layout for simple forms where form fields must span across the available width (`horizontalConstraint: scale`).
```

Reviewer: Got it, good to know 🤗
